cassandra/common/src/main/java/org/opennms/newts/cassandra/CassandraSessionImpl.java

```diff
@@ -15,115 +15,144 @@
         if (coreConnectionsPerHost != null) {
-            LOG.debug("Using {} core connections per host.", coreConnectionsPerHost);
-            poolingOptions.setCoreConnectionsPerHost(HostDistance.LOCAL, coreConnectionsPerHost)
-                    .setCoreConnectionsPerHost(HostDistance.REMOTE, coreConnectionsPerHost);
+            LOG.warn("Ignoring core connections per host value of: {}. " +
+                    "Use 'max-connections-per-host' only instead.", coreConnectionsPerHost);
+
         }
         if (maxConnectionsPerHost != null) {
             LOG.debug("Using {} max connections per host.", maxConnectionsPerHost);
-            poolingOptions.setMaxConnectionsPerHost(HostDistance.LOCAL, maxConnectionsPerHost)
-                    .setMaxConnectionsPerHost(HostDistance.REMOTE, maxConnectionsPerHost);
+            configBuilder.withInt(DefaultDriverOption.CONNECTION_POOL_LOCAL_SIZE, maxConnectionsPerHost);
+            configBuilder.withInt(DefaultDriverOption.CONNECTION_POOL_REMOTE_SIZE, maxConnectionsPerHost);
         }
         if (maxRequestsPerConnection != null) {
             LOG.debug("Using {} max requests per connection.", maxRequestsPerConnection);
-            poolingOptions.setMaxRequestsPerConnection(HostDistance.LOCAL, maxRequestsPerConnection)
-                    .setMaxRequestsPerConnection(HostDistance.REMOTE, maxRequestsPerConnection);
+            configBuilder.withInt(DefaultDriverOption.CONNECTION_MAX_REQUESTS, maxRequestsPerConnection);
         }
-
-        Builder builder = Cluster
-                .builder()
-                .withPort(port)
-                .addContactPoints(hostname.split(","))
-                .withReconnectionPolicy(new ExponentialReconnectionPolicy(1000, 2 * 60 * 1000))
-                .withPoolingOptions(poolingOptions)
-                .withCompression(Compression.valueOf(compression.toUpperCase()));
+        configBuilder.withClass(DefaultDriverOption.RECONNECTION_POLICY_CLASS, ExponentialReconnectionPolicy.class);
+        configBuilder.withDuration(DefaultDriverOption.RECONNECTION_BASE_DELAY, Duration.of(1, ChronoUnit.SECONDS));
+        configBuilder.withDuration(DefaultDriverOption.RECONNECTION_MAX_DELAY, Duration.of(2, ChronoUnit.MINUTES));
+        configBuilder.withString(DefaultDriverOption.PROTOCOL_COMPRESSION, compression.toUpperCase());
 
         if (username != null && password != null) {
             LOG.info("Using username: {} and password: XXXXXXXX", username);
-            builder.withCredentials(username, password);
+            configBuilder.withString(DefaultDriverOption.AUTH_PROVIDER_USER_NAME, username);
+            configBuilder.withString(DefaultDriverOption.AUTH_PROVIDER_PASSWORD, password);
         }
 
         if (ssl) {
-            LOG.info("Enabling SSL.");
-            builder.withSSL();
+            throw new IllegalArgumentException("Driver must be configured with \"cassandra.driver-settings-file\" to enable SSL support.");
```

Comment:
Here too, if an option has to be done a different way, why not get rid of it altogether?